sources/packages/backend/apps/workers/src/controllers/assessment/assessment.controller.ts

```diff
@@ -271,6 +272,28 @@ export class AssessmentController {
           );
           return job.complete();
         }
+        const application = assessment.application;
+        if (
+          application.applicationStatus === ApplicationStatus.Completed &&
+          !assessment.previousDateChangedReportedAssessment
+        ) {
+          const updatedLastReportedAssessment =
+            await this.studentAssessmentService.updateLastReportedAssessment(
+              job.variables.assessmentId,
+              entityManager,
+            );
+          if (updatedLastReportedAssessment?.affected) {
+            jobLogger.log("Last reported date changed assessment was updated.");
+          } else if (updatedLastReportedAssessment === undefined) {
+            jobLogger.log(
+              "No previous reported assessment found or no last reported assessment update needed.",
+            );
+          } else if (!updatedLastReportedAssessment) {
+            jobLogger.log("Last reported assessment was previously updated.");
```

Comment:
This log is not required now.

Reply:
Don't we need it for the idempotency scenario. However, I should update the `else if` to `else if (!lastReportedAssessmentUpdateResult?.affected)`.

Please let me know if this works.